### DISCUSSION TITLE
Running showcreator on linux

### DISCUSSION THREAD

**Discussion by worldpeace-germany** [missionpinball/missionpinball#109] [Q&A]
Hello,

I am not able to get the showcreator to work on Linux (using openSuse). I am following https://missionpinball.org/tools/showcreator/ where it tells me to run install.bat. Looking into the bat file I see that at the end of the day it would like to execute some tool call bmk from the bin directory. When I go into the bin directory I see a bmk file, when I try to execute (regardless via the install.bat or just from the command line) I am getting an error from the bash that this file or folder is not being found. It is there, even autocompletion of the bash finds it.  Any idea what might be wrong? What is that bmk tool, I tried to google this tool but basically didn't find anything.

Thanks.

**Comment by toomanybrians** [comment]
I’ve never used the showcreator and don’t have a linux vm handy to try, so I’m just guessing here.

Looking at the repo mentioned in the linux install instructions, I wonder if it’s simply something like the MPF instructions are old? I wonder if you can follow the ubuntu instructions from blitzmax and get it installed that way?

Or can you post the terminal results of running that batch file? I know that permissions could prevent it from being seen. (I notice the install.bat file does use `chmod +x` to give some things execute rights.

If I was doing this on my own, I would use ChatGPT. Dunno if you’ve tried that for tech support. I would copy and paste in the entire MPF doc page for installing on linux, tell it you’re trying to install this, following those instructions, and then paste in the entire results of terminal session with the errors. Honestly ChatGPT is my #1 troubleshooting partner now. 🤣

**Comment by worldpeace-germany** [comment]
execution rights is not the problem, I tried to follow the Ubuntu instructions as much as possible, but it would just not run these files. Maybe when I have some time I will spin up an Ubuntu VM to see if on Ubuntu it is working. That might give a hint.

**Comment by worldpeace-germany** [comment]
I spun up an Ubuntu VM to see if that works there. I doesn't, I get compile issue need some header file Xft.h  for that you need do to a apt get install libxft-dev, now I am getting ft2build.h is missing. What I (=Google) figured out is that it comes as part as some truetype package, I have installed (libfreetype6-dev) it. But that error I am not able to get resolved. If someone has an Ubuntu and the passion to look into it, then that is highly appreciated.

**Comment by worldpeace-germany** [comment]
Just in case someone comes across this post, see https://github.com/orgs/missionpinball/discussions/112 for an alternative.

**Comment by worldpeace-germany** [comment]
Okay, I got a bit further, but not sure if I am on the right track.

This

`apt install linux-libc-dev:i386 libxft2:i386 g++-multilib gcc-multilib libxpm-dev:i386 libxxf86vm-dev:i386 libgl1-mesa-dev:i386 libglu1-mesa-dev:i386`

doesn't work for me, I had to remove all :i386, then I was able to install it. The blitzmax repro says you need 32-bit libs, so if that is true it seems they don't exist anymore, at the same time the blitzmax repro is very old, not sure if still maintained.

After that I still get some header files missing, among them ft2build.h as mentioned above already, I figured out a export `CPLUS_INCLUDE_PATH="/usr/include/freetype2" `fixed that problem.

Now I was able to compile the full blitzmax thingy, at least I don't get an error.... Now the makeapp led.bmx, when running this I get

`Unable to convert from Int to Byte Ptr led.mx;1646;2`

It seems that this programming language is Blitzmax and it seems not being too much supported https://en.wikipedia.org/wiki/Blitz_BASIC article shows http://www.blitzbasic.com/ which is not existing anymore. Anyways the line in question about shows

`Local zero:Byte Ptr = 0`

According to https://blitzmax.org/docs/en/api/brl/brl.blitz/#byte I would say that line is okay.

This code
```

	Local hdir:Byte Ptr
	Local zero:Byte Ptr = 0
	
	s = 0
	hdir = ReadDir(dir$)
	If hdir <> zero
```

I have changed to

```
	'Local hdir:Byte Ptr
	'Local zero:Byte Ptr = 0
	
	s = 0
	Local hdir:Byte = ReadDir(dir$)
	If hdir <> Null
```

This brings me past that error. Now it brings up the next error about incompatible libX11.so and whatsoever... Anyone an idea?

**Comment by worldpeace-germany** [comment]
PS: Cross reference to showcreator github issues https://github.com/missionpinball/showcreator/issues/3

**Comment by worldpeace-germany** [comment]
Found a way of getting it to work, pull request for documentation update https://github.com/missionpinball/mpf-docs/pull/490
